components/site/src/lib.rs

```diff
@@ -799,19 +799,30 @@ impl Site {
     }
 
     fn index_for_lang(&self, lang: &str) -> Result<()> {
-        let index_json = search::build_index(lang, &self.library.read().unwrap(), &self.config)?;
-        let (path, content) = match &self.config.search.index_format {
-            IndexFormat::ElasticlunrJson => {
-                let path = self.output_path.join(format!("search_index.{}.json", lang));
-                (path, index_json)
+        let extension = match &self.config.search.index_format {
+            IndexFormat::ElasticlunrJavascript | IndexFormat::FuseJavascript => "js",
+            IndexFormat::ElasticlunrJson | IndexFormat::FuseJson => "json",
+        };
+        let path = &self.output_path.join(format!("search_index.{}.{}", lang, extension));
+        let library = self.library.read().unwrap();
+        let content = match &self.config.search.index_format {
+            IndexFormat::ElasticlunrJavascript | IndexFormat::ElasticlunrJson => {
+                search::build_index(lang, &library, &self.config)?
```

Comment:
we should probably put elastic search into its own file to keep it symmetric

Reply:
done